lib/analysis.js

```diff
@@ -52,6 +53,7 @@ function AnalysisFactory(DatabaseServiceClass) {
 
 AnalysisFactory.prototype.create = function(configuration, definition, callback) {
     debug('Using configuration=%j', configuration);
+    var logger = createLogger(configuration.logger).child({ analysis: definition.type, user: configuration.user });
```

Comment:
Ok. So we'll need a mechanism to connect different entries in log that belong to the same analysis operation. We can achieve that: 
 * creating an internal _transaction id_ and use it to log everything of the same analysis 
 * passing to `AnalysisFactory.create` a _transaction id_ and use it to log.